### DISCUSSION TITLE
Picker shows medium image : how to override it ?

### DISCUSSION THREAD

**Discussion by christopher87** [awcodes/filament-curator#251] [General]
Hi,

I am using latest stable version 2.8.5, with Filament 2.

I have uploaded the following image :
![image](https://i.ibb.co/9vSts1J/Screenshot-2023-08-04-at-22-38-47.jpg)

The picker shows a cropped image (format "medium_url") :

- With constrained(true) :
![image](https://i.ibb.co/mXJTbJQ/Screenshot-2023-08-04-at-22-39-10.jpg)

- With constrained(false) :
![image](https://i.ibb.co/xDfHPZ7/Screenshot-2023-08-04-at-22-38-57.jpg)

But I wish I could display image not cropped.
How can I override this behaviour ?

Thanks in advance,

**Comment by howdu** [comment]
You could create a custom Media model and modify the medium image attribute.

```php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Casts\Attribute;

class Media extends \Awcodes\Curator\Models\Media
{
    protected function mediumUrl(): Attribute
    {
        return Attribute::make(
            get: fn () => $this->getSignedUrl(['w' => 1000, 'h' => 260, 'fit' => 'contain', 'fm' => 'webp']),
        );
    }
}
```

Then from your app service provider tell Curator to use this model - https://github.com/awcodes/filament-curator#global-settings
```php
use Awcodes\Curator\Facades\Curator;

public function register()
{
    Curator::mediaModel(\App\Models\Media::class);
}
```

**Reply by christopher87** [reply]
Hello,

Thank you it works like a charm !

That was my first idea, but maybe it could be overridden with a config file in a future release ?

By the way it works very well, thank you again :)

PS : I don't know how to mark your answer as "green answer" ?
